File: Backend/fix_prompt.py
with open('src/ai/ai.service.ts', 'rb') as f:
    content = f.read()

# Find and replace the double-plus bug
old = b"QCM techniques pour ce poste:\\\\n' + +"
new  = b"QCM techniques pour ce poste:\\\\n' +"

if old in content:
    content = content.replace(old, new)
    with open('src/ai/ai.service.ts', 'wb') as f:
        f.write(content)
    print('FIXED - double plus removed')
else:
    # Try alternate forms
    old2 = "QCM techniques pour ce poste:\\n' + +".encode('utf-8')
    if old2 in content:
        content = content.replace(old2, "QCM techniques pour ce poste:\\n' +".encode('utf-8'))
        with open('src/ai/ai.service.ts', 'wb') as f:
            f.write(content)
        print('FIXED - alternate form')
    else:
        # Show what's around that area
        idx = content.find(b'pour ce poste')
        if idx >= 0:
            print('Found at:', idx)
            print('Context:', repr(content[idx:idx+60]))
        else:
            print('Pattern not found at all')

File: Backend/fix_prompt.js
const fs = require('fs');
const path = require('path');
const filePath = path.join(__dirname, 'src', 'ai', 'ai.service.ts');

let content = fs.readFileSync(filePath, 'utf8');

// Show context around the problem area
const idx = content.indexOf("pour ce poste");
if (idx < 0) {
  console.log('ERROR: pattern not found');
  process.exit(1);
}
console.log('Context:', JSON.stringify(content.slice(idx, idx + 80)));

// Fix: remove the extra + sign
const badPattern = "pour ce poste:\\n' + +\r\n      jobDescription";
const goodPattern = "pour ce poste:\\n' +\r\n      jobDescription";
if (content.includes(badPattern)) {
  content = content.replace(badPattern, goodPattern);
  fs.writeFileSync(filePath, content, 'utf8');
  console.log('FIXED!');
} else {
  // Try without \r
  const badPattern2 = "pour ce poste:\\n' + +\n      jobDescription";
  if (content.includes(badPattern2)) {
    content = content.replace(badPattern2, "pour ce poste:\\n' +\n      jobDescription");
    fs.writeFileSync(filePath, content, 'utf8');
    console.log('FIXED (LF only)!');
  } else {
    console.log('Pattern not found. Content snippet:', JSON.stringify(content.slice(idx, idx + 100)));
  }
}
